chronos_bin/src/telemetry/otlp_collector.rs

```diff
@@ -0,0 +1,40 @@
+use opentelemetry::trace::TraceError;
+use opentelemetry::{
+    global,
+    sdk::{propagation::TraceContextPropagator, trace as sdktrace},
+};
+use opentelemetry_otlp::{Protocol, WithExportConfig};
+
+pub struct OtlpCollector {}
+
+impl Default for OtlpCollector {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl OtlpCollector {
+    pub fn new() -> Self {
+        OtlpCollector {}
+    }
+
+    pub fn http_collector_connect(&self, protocol: Protocol) -> Result<sdktrace::Tracer, TraceError> {
+        // service name will be picked from  "OTEL_SERVICE_NAME" env variable
+
+        if let Ok(trace_exporter) = std::env::var("OTEL_EXPORTER_OTLP_TRACES_ENDPOINT") {
+            global::set_text_map_propagator(TraceContextPropagator::new());
+            opentelemetry_otlp::new_pipeline()
+                .tracing()
+                .with_exporter(opentelemetry_otlp::new_exporter().http().with_protocol(protocol).with_endpoint(trace_exporter))
```

Comment:
We are taking protocol as parameter but at the same time calling `.http()`. Is this intentional?

Reply:
Yes right now only http exporter is available as a Collector Daemon in a Kubelet, I could also provision GRPC but don't have a live environment so left it for future ticket.